aws/rust-runtime/aws-config/src/profile/parser.rs

```diff
@@ -114,6 +114,11 @@ impl ProfileSet {
         self.profiles.get(profile_name)
     }
 
+    /// Set a named profile in the profile set
```

Comment:
let's add docs that clarify that this will replace another profile with the same name. Might as well change the API to return the removed profile if it existed as well